### PR TITLE
Redux toolkit tutorial added

### LINKED ISSUE #326: [Feature Request]: Redux toolkit full tutorial
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

i want to add redux toolkit full tutorial in react course advnaced level
i will use online resource and make full structured redux toolkitcourse

### Use Case

1.  i will add multiple lessons
2. images or better understanding
3. give roadmap in first lesson
4. making concept clear

### Benefits

will add one more section for react course

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by Yashgabani845**
## Related Issue

Fixes #326 
## Description

 i have added 3 lessons in redux toolkit tutorial 
 evry lesson has these sections

-  introduction
-  steps
- required thing to know
- basic codes
- desrption of the code
- image/GIF to understand it better way
- conslusion

## Type of PR

- [x] Documentation update

## Screenshots / Videos (if applicable)

https://github.com/CodeHarborHub/codeharborhub/assets/127788321/fd791a22-a92b-4b8c-9096-9799c0274b29

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

if you're impressed with my work then please upgrade PR level, 
and if any changes needed please tell me 
always respect your feedback
thankyou

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

**Comment by Yashgabani845** [comment]
@Ajay-Dhangar  
redux-toolkit done
please review this PR and feel free to give feedback and suggestions

File: courses/react-js/advanced-level/redux-toolkit/lesson_1.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+```bash
+npm install @reduxjs/toolkit react-redux
+```
+
+### 2. Creating a Redux Store
+
+Redux Toolkit provides a `configureStore` function that simplifies store creation. It includes good defaults and automatically sets up the Redux DevTools extension.
+
+```javascript
+// src/app/store.js
```
Comment (Ajay-Dhangar) [inline courses/react-js/advanced-level/redux-toolkit/lesson_1.md:46]: remove `// src/app/store.js` and use like 

````md
```Javascript title="src/app/store.js"
// your code here
```
````

Comment (Ajay-Dhangar) [reply courses/react-js/advanced-level/redux-toolkit/lesson_1.md:46]: and apply for all code

File: courses/react-js/advanced-level/redux-toolkit/lesson_1.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+});
+
+export default store;
+```
+
+### 3. Creating a Slice
+
+A slice is a collection of Redux reducer logic and actions for a single feature of your app. Redux Toolkit’s `createSlice` function automatically generates action creators and action types.
+
+```javascript
```
Comment (Ajay-Dhangar) [inline courses/react-js/advanced-level/redux-toolkit/lesson_1.md:63]: remove `// src/features/counter/counterSlice.js` and use like 

````md
```Javascript title="src/features/counter/counterSlice.js"
// your code here
```
````

File: courses/react-js/advanced-level/redux-toolkit/lesson_1.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+export const { increment, decrement, incrementByAmount } = counterSlice.actions;
+export default counterSlice.reducer;
+```
+
+### 4. Using the Redux Store in a React Component
+
+Now that the store and slice are set up, you can use them in your React components. Use the `useSelector` hook to read state and the `useDispatch` hook to dispatch actions.
+
+```javascript
```
Comment (Ajay-Dhangar) [inline courses/react-js/advanced-level/redux-toolkit/lesson_1.md:93]: use `title`

File: courses/react-js/advanced-level/redux-toolkit/lesson_1.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+export default Counter;
+```
+
+### 5. Providing the Redux Store to Your Application
+
+To make the Redux store available to your entire app, wrap your application in the `Provider` component from `react-redux`.
+
+```javascript
```
Comment (Ajay-Dhangar) [inline courses/react-js/advanced-level/redux-toolkit/lesson_1.md:124]: use `title`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
everything is fine but use `title` in code for best practice

**Comment by Yashgabani845** [comment]
@Ajay-Dhangar  i have chnaged required titles

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**